.github/CONTRIBUTING.md

```diff
@@ -26,6 +26,10 @@ Contributions via pull requests are much appreciated. Before sending us a pull r
 1. You are working against the latest source on the *main* branch.
 2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
 3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
+4. The size of the PR is small. If the issue requires sizeable code change, you can go with the following:
```

Comment:
Minor - I think not breaking the tip of the main branch is also important if the changes are split just because size is large, if we can be clear it will be great , but not a major thing as it is covered by - I have tested my changes from the checklist.

Reply:
Yes, changed to 
`Break down the PR into independent components(that do not break the library functionality) and create separate PRs.`